### PR TITLE
Fix 'Failed to create or update nsm' error in log on deploying infra

### PR DESCRIPTION
**Pull request by denis-tingajkin**
Signed-off-by: denis-tingajkin <denis.tingajkin@xored.com>

## Description

Fix for https://github.com/networkservicemesh/networkservicemesh/issues/991
## How Has This Been Tested?

- [ ] Covered by existing integration testing
- [X] Added unit testing to cover
- [X] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: k8s/pkg/registryserver/registry_cache.go [hunk 1]
```diff
@@ -46,6 +47,7 @@ func NewRegistryCache(clientset *nsmClientset.Clientset) RegistryCache {
 		clientset:                   clientset,
 		stopFuncs:                   make([]func(), 0, 3),
 		nsmNamespace:                namespace.GetNamespace(),
+		createOrUpdate:              sync.Mutex{},
```
Comment (haiodo) [inline k8s/pkg/registryserver/registry_cache.go]: No need to create it. It is already defined at line 39

File: k8s/pkg/registryserver/resource_cache/ns_cache.go [hunk 2]
```diff
@@ -3,16 +3,17 @@ package resource_cache
...
 type NetworkServiceCache struct {
 	cache           abstractResourceCache
-	networkServices map[string]*v1.NetworkService
+	networkServices sync.Map
 }
 
 func NewNetworkServiceCache() *NetworkServiceCache {
 	rv := &NetworkServiceCache{
-		networkServices: make(map[string]*v1.NetworkService),
+		networkServices: sync.Map{},
```
Comment (haiodo) [inline k8s/pkg/registryserver/resource_cache/ns_cache.go]: No need to override value from structure.

Comment (denis-tingajkin) [reply k8s/pkg/registryserver/resource_cache/ns_cache.go]: oops! Thanks, fixed

File: k8s/pkg/registryserver/registry_cache.go (hunk 1 above)
Comment (denis-tingajkin) [reply k8s/pkg/registryserver/registry_cache.go]: oops! Thanks, fixed

**Comment by haiodo** [comment]
+1

**Pull request merged**